src/hardhat/compiler/index.ts

```diff
@@ -130,49 +122,22 @@ subtask(
       return error
     })
 
-    // Filter out any "No input sources specified" errors, but only if one of the two compilations
-    // threw the error. Basically, it might be intended for only one of the EVM or OVM compilers to
-    // be compiling contracts, but something went wrong if *both* compilers recieve no input.
-    let errors = (ovmOutput.errors || []).concat(evmOutput.errors || [])
-    const filtered = errors.filter((error: any) => {
-      return error.message !== 'No input sources specified.'
-    })
-
-    // Make sure we only saw one of those "No input sources specified." errors.
-    if (errors.length === filtered.length + 1) {
-      errors = filtered
-    }
-
-    // Transfer over any OVM outputs to the EVM output, with an identifier.
-    for (const fileName of Object.keys(ovmOutput.contracts || {})) {
-      if (Object.keys(evmOutput.contracts || {}).includes(fileName)) {
-        for (const contractName of Object.keys(ovmOutput.contracts[fileName])) {
-          const contractOutput = ovmOutput.contracts[fileName][contractName]
-
-          // Need to fix any link references in the OVM outputs. Otherwise we'll be trying to link
-          // an OVM-compiled contract to EVM-compiled contracts.
-          const linkRefs = contractOutput.evm?.bytecode?.linkReferences
-          for (const linkRefFileName of Object.keys(linkRefs || {})) {
-            for (const [linkRefName, linkRefOutput] of Object.entries(
-              linkRefs[linkRefFileName]
-            )) {
-              delete linkRefs[linkRefFileName][linkRefName]
-              linkRefs[linkRefFileName][`${linkRefName}-ovm`] = linkRefOutput
-            }
-          }
-
-          // OVM compiler output is signified by adding -ovm to the output name.
-          evmOutput.contracts[fileName][`${contractName}-ovm`] = contractOutput
-        }
-      }
-    }
```

Comment:
Note to reader: This can be removed because we're adding a new `artifacts-ovm` path to store them there.